### ISSUE TITLE
Configurable product on wishlist shows parent image instead variation image

### ISSUE THREAD

**Issue by anebi** [magento/magento2#8168]
### Preconditions

1. Magento 2.1.3
2. PHP7
3. MySQL 5.7.16 

### Steps to reproduce
1. Create product from Configurable type with default image
2. Assign products to this Configurable with let's say color and size attributes. Let the associated products have different images
3. Go to that configurable product on frontend, select configurations and add the product to wishlist.
4. Go to wishlist in my account and you will notice that wishlist displays images of parent product instead of the product that represent selected configurations

### Expected result

1. Wishlist should display variation product image instead of parent
2. Currently this is done for checkout page and cart and we are able to select from backend if we want parent or child product image to be displayed

**Comment by Szymon-GP** [comment]
I have the same issue, if you add different variations they will be listed separately in whishlist but the image is wrong, it represents configurable product rather than specific variation

**Comment by naydav** [comment]
Internal ticket MAGETWO-8709 which tracks this GitHub issue, is in our issue backlog.
Thanks

**Comment by sbrandollo** [comment]
If someone is interesting to a workaround, I suggest creating a plugin for 
Magento\Wishlist\Block\Customer\Wishlist\Item\Column\Image
and write something like:
```
public function aroundGetImage(
        Image $subject,
        \Closure $proceed,
        Product $product,
        $imageId,
        $attributes
    ) {
        if ($product->getTypeId() == 'configurable') 
        {
            $simpleProduct = $this->productRepository->get($product->getSku());
            $url = $this->config->getBaseMediaUrl() . $simpleProduct->getImage();
            $block = $proceed($product, $imageId, $attributes);
            $block->setImageUrl($url);
            return $block;
        }
        return $proceed($product, $imageId, $attributes);
    }
```

**Comment by magento-engcom-team** [comment]
@anebi, thank you for your report.
We've created internal ticket(s) MAGETWO-8709 to track progress on the issue.

**Comment by RomaKis** [comment]
Working on this.

**Comment by magento-engcom-team** [comment]
Hi @anebi. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1031 by @RomaKis in 2.2-develop branch
Related commit(s):
 - [93a5c391495399a59f5ee922718890820e601d67](https://github.com/magento/magento2/commit/93a5c391495399a59f5ee922718890820e601d67)

The fix will be available with the upcoming 2.2.5 release.

**Comment by cpartica** [comment]
Note that this fix is different on 2.3. Wishlist should obey cart image setting from config. Fix will be backported to 2.2 future releases

**Comment by dhrumin44** [comment]
It was fixed in new **2.2.6** release version, Please check it once from here [pull/15477](https://github.com/magento/magento2/pull/15477).

**Comment by codewidsaurav** [comment]
i face this issue in magento 2.4 , in customer > my wishlist --> it shows configurable product image instead of selected variant
